File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/python/path_traversal_string_literal_fp.py
"""
vt=PathTraversal lang=python — names below are inside strings/comments only

Documentation example (these patterns are NOT executed):
  x = request.args.get("id")
  cursor.execute("SELECT ... " + x)
  os.system(x)
  open(x, "r")
  requests.get(x)
  pickle.loads(x)
  response.write(x)
"""

DOC = """
request.args, cursor.execute, os.system, open(, requests.get, pickle.loads, response.write
"""

def docs_only():
    # also: request.args.get("id") + cursor.execute(...) — comment only
    msg = "request.args.get + cursor.execute + os.system + open( + requests.get + pickle.loads"
    return msg

File: crates/tldr-cli/tests/fixtures/vuln_migration_v1/java/path_traversal_string_literal_fp.java
// vt=PathTraversal lang=java — names below are inside strings/comments only
// request.getParameter is a source; stmt.executeQuery, Runtime.getRuntime().exec,
// new java.io.File, new java.net.URL, new java.io.ObjectInputStream are sinks.

public class DocsOnly {
    public String docs() {
        String doc = "request.getParameter -> stmt.executeQuery(SELECT ... )";
        String more = "Runtime.getRuntime().exec, new java.io.File, new java.net.URL, ObjectInputStream";
        return doc + more;
    }
}
